prefect_kubernetes/jobs.py

```diff
@@ -299,3 +309,205 @@ def kubernetes_orchestrator():
             namespace=namespace,
             **kube_kwargs,
         )
+
+
+class KubernetesJobRun(JobRun[Dict[str, Any]]):
+    """A container representing a run of a Kubernetes job."""
+
+    def __init__(self, kubernetes_job: "KubernetesJob"):
+        self._kubernetes_job = kubernetes_job
+        self.pod_logs = {}
+
+    async def wait_for_completion(self):
+        """Waits for the job to complete.
+
+        Raises:
+            RuntimeError: If the Kubernetes job fails.
+        """
+        with self._kubernetes_job.credentials.get_client(
+            "batch"
+        ) as batch_v1_client, self._kubernetes_job.credentials.get_client(
+            "core"
+        ) as core_v1_client:
+            completed = False
+
+            elapsed_time = 0
+
+            while not completed:
+                job_expired = (
+                    elapsed_time > self._kubernetes_job.timeout_seconds
+                    if self._kubernetes_job.timeout_seconds
+                    else False
+                )
+                if job_expired:
+                    raise TimeoutError(f"Job timed out after {elapsed_time} seconds.")
```

Comment:
Alex recommends using a separate exception because this timeout error is: "Raised when a system function timed out at the system level"